### PR TITLE
Scala 2.12.0-M1

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

**Comment by lrytz** [comment]
Review by @retronym @sjrd

File: download/_posts/2015-05-05-2.12.0-M1.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Scala 2.12.0-M1
+start: 05 May 2015
+layout: downloadpage
+release_version: 2.12.0-M1
+release_date: "May 05, 2015"
+show_resources: "true"
+permalink: /download/2.12.0-M1.html
+requirements: "This Scala software distribution can be installed on any Unix-like or Windows system. It requires the Java runtime version 1.6 or later, which can be downloaded <a href='http://www.java.com/'>here</a>."
```
Comment (retronym) [inline download/_posts/2015-05-05-2.12.0-M1.md]: Could you please add a note to the release checklist for M2 to remind us to edit this to require 1.8?

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+
+### Binary Compatibility
+
+When two versions of Scala are binary compatible, it is safe to compile your project on one Scala version and link against another Scala version at run time.
+Safe run-time linkage (only!) means that the JVM does not throw a (subclass of) [`LinkageError`](http://docs.oracle.com/javase/7/docs/api/java/lang/LinkageError.html) when executing your program in the mixed scenario, assuming that none arise when compiling and running on the same version of Scala.
+Concretely, this means you may have external dependencies on your run-time classpath that use a different version of Scala than the one you're compiling with, as long as they're binary compatibile.
+In other words, separate compilation on different binary compatible versions does not introduce problems compared to compiling and running everything on the same version of Scala.
+
+We check binary compatibility automatically with [MiMa](https://github.com/typesafehub/migration-manager).
```
Comment (retronym) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: Maybe for next release we should refactor the bulk of the BC verbiage to a dedicated page.

**Comment by retronym** [comment]
LGTM, thanks for pulling all of this together.

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Notable examples where we know meta-binary compatibility is hard to achieve: specialisation and the optimizer.
+
+In short, if binary compatibility of your library is important to you, use [MiMa](https://github.com/typesafehub/migration-manager) to verify compatibility before releasing.
+Compiling identical sources with different versions of the Scala compiler (or on different JVM versions!) could result in binary incompatible bytecode.
+This is rare, and we try to avoid it, but we can't guarantee it will never happen.
+
+#### Concretely
+
+We guarantee forwards and backwards compatibility of the `"org.scala-lang" % "scala-library" % "2.12.x"` and `"org.scala-lang" % "scala-reflect" % "2.12.x"` artifacts, except for anything under the `scala.reflect.internal` package, as scala-reflect is still experimental.
+We also strongly discourage relying on the stability of `scala.concurrent.impl` and `scala.reflect.runtime`, though we will only break compatibility for severe bugs here.
```
Comment (sjrd) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: This is misleading for milestones. We don't actually guarantee any kind of binary compatibility between milestones, do we?

File: download/_posts/2015-05-05-2.12.0-M1.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  ["-main-unixsys", "scala-2.12.0-M1.tgz", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-2.12.0-M1.tgz", "Max OS X, Unix, Cygwin", "23.85M"],
+  ["-main-windows", "scala-2.12.0-M1.msi", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-2.12.0-M1.msi", "Windows (msi installer)", "102.77M"],
+  ["-non-main-sys", "scala-2.12.0-M1.zip", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-2.12.0-M1.zip", "Windows", "23.90M"],
+  ["-non-main-sys", "scala-2.12.0-M1.deb", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-2.12.0-M1.deb", "Debian", "70.35M"],
+  ["-non-main-sys", "scala-2.12.0-M1.rpm", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-2.12.0-M1.rpm", "RPM package", "101.61M"],
+  ["-non-main-sys", "scala-docs-2.12.0-M1.txz", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-docs-2.12.0-M1.txz", "API docs", "43.72M"],
+  ["-non-main-sys", "scala-docs-2.12.0-M1.zip", "http://downloads.typesafe.com/scala/2.12.0-M1/scala-docs-2.12.0-M1.zip", "API docs", "81.12M"],
+  ["-non-main-sys", "scala-sources-2.12.0-M1.tar.gz", "https://github.com/scala/scala/archive/v2.12.0-M1.tar.gz", "Sources", ""]
+]
+---
```
Comment (sjrd) [inline download/_posts/2015-05-05-2.12.0-M1.md]: I don't remember whether we're that strict with this repo, but there should be an EOL at the end of file, here.

File: download/index.md [hunk 5]
```diff
@@ -6,7 +6,8 @@ release_date: "March 5, 2015"
 other_releases: [
   ["maintenance_version", "Last 2.9.x maintenance release", 2.9.3, "February 28, 2013"],
   ["maintenance_version", "Last 2.10.x maintenance release", 2.10.5, "March 5, 2015"],
-  ["development_version", "Current 2.11.x release", 2.11.6, "March 5, 2015"]
+  ["development_version", "Current 2.11.x release", 2.11.6, "March 5, 2015"],
+  ["milestone_version", "Current 2.12.x milestone release", 2.12.0-M1 "May 5, 2015"]
```
Comment (sjrd) [inline download/index.md]: Missing `,` after `2.12.0-M1`

File: download/_posts/2015-05-05-2.12.0-M1.md (hunk 1 above)
Comment (lrytz) [inline download/_posts/2015-05-05-2.12.0-M1.md]: ok, sure!

File: download/index.md (hunk 5 above)
Comment (lrytz) [inline download/index.md]: thanks!

File: download/_posts/2015-05-05-2.12.0-M1.md (hunk 4 above)
Comment (lrytz) [inline download/_posts/2015-05-05-2.12.0-M1.md]: the file was generated by https://github.com/scala/make-release-notes.. i'll see if I can easily change it.

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md (hunk 3 above)
Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: that's right, thanks! i'll clarify.

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md (hunk 2 above)
Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: agreed.

**Comment by sjrd** [comment]
That's all from me.

**Comment by sjrd** [comment]
Oh `_config.yml` should also be updated, I believe.

**Comment by lrytz** [comment]
i didn't see what to change in [`_config.yml`](https://github.com/scala/scala-lang/blob/master/_config.yml) - or maybe the `devscalaversion`?

**Comment by sjrd** [comment]
Yes, according to its history, `devscalaversion` used to hold Milestones.

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md (hunk 3 above)
Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: added here: https://github.com/scala/scala-lang/pull/298/files#diff-7870b8e60365cd372d5f6130e741ea6cR131

**Comment by lrytz** [comment]
All comments addressed, terminating newlines added.

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+### Available Projects
+
+Please refer to the list of [libraries and frameworks available for Scala 2.12](https://github.com/scala/make-release-notes/blob/2.12.x/projects-2.12.md).
+
+
+### Binary Compatibility
+
+Since Sala 2.11, minor releases of Scala are binary compatible.
+Scala 2.12 continues this tradition: every 2.12.x release will be binary compatible to 2.12.0.
+Note that milestone releases and release candidates are **not** binary compatible to any other release.
```
Comment (sjrd) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: compatible <del>to</del><ins>with</ins>?

Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: ok :)

Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: done

**Comment by sjrd** [comment]
LGTM

File: news/_posts/2015-05-05-release-notes-2.12.0-M1.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+
+### New Features in the 2.12 Series
+
+Scala 2.12 contains all of the bug fixes and improvements made in the 2.11 series.
+
+At the current stage in the milestone cycle, Scala 2.12 is still very similar to Scala 2.11.
+
+The following changes are planned for Scala 2.12:
+  * Java 8 style closures.
```
Comment (sjrd) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: These bullets don't render correctly on the Website :-(
http://www.scala-lang.org/news/2.12.0-M1

Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: yes, thanks.. fixing it

Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: hmm... i pushed a new commit that fixed it when running locally (`jekyll serve`), but it seems that didn't work: http://scala-lang.org/news/2.12.0-M1

Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: i'll try with removing the indentation of the bullet list.

Comment (lrytz) [inline news/_posts/2015-05-05-release-notes-2.12.0-M1.md]: ok, that worked, now it looks good.

**Pull request merged**